packages/cloud_firestore/cloud_firestore/lib/src/utils/codec_utility.dart

```diff
@@ -0,0 +1,47 @@
+part of cloud_firestore;
+
+class _CodecUtility {
+  static Map<String, dynamic> _replaceValueWithDelegatesInMap(
+      Map<String, dynamic> data) {
+    Map<String, dynamic> output = Map.from(data);
+    output.updateAll((_, value) => _valueEncode(value));
+    return output;
+  }
+
+  static List<dynamic> _replaceValueWithDelegatesInArray(List<dynamic> data) {
+    return List.from(data).map((value) => _valueEncode(value));
+  }
+
+  static Map<String, dynamic> _replaceDelegatesWithValueInMap(
+      Map<String, dynamic> data) {
+    Map<String, dynamic> output = Map.from(data);
+    output.updateAll((_, value) => _valueDecode(value));
+    return output;
+  }
+
+  static List<dynamic> _replaceDelegatesWithValueInArray(List<dynamic> data) {
+    return List.from(data).map((value) => _valueDecode(value));
+  }
+
+  static dynamic _valueEncode(dynamic value) {
```

Comment:
This shouldn't be prefixed by _ so it's "public". (Similar to the _PlatformUtils)